src/engine/server_iv.c

```diff
@@ -609,7 +615,7 @@ ivc_on_get(crt_iv_namespace_t ivns, crt_iv_key_t *iv_key,
 	D_ALLOC_PTR(priv_entry);
 	if (priv_entry == NULL) {
 		class->iv_class_ops->ivc_ent_put(entry, entry_priv_val);
-		D_GOTO(out, rc);
+		D_GOTO(out, rc = -DER_NOMEM);
```

Comment:
This has it's own ticket and is also included in https://github.com/daos-stack/daos/pull/12635 which should land first.